pkg/controller/install/deployment.go

```diff
@@ -224,6 +160,16 @@ func (i *StrategyDeploymentInstaller) checkForDeployments(deploymentSpecs []Stra
 		if !ready {
 			return StrategyError{Reason: StrategyErrReasonWaiting, Message: fmt.Sprintf("waiting for deployment %s to become ready: %s", dep.Name, reason)}
 		}
+
+		// check annotations
+		if len(i.templateAnnotations) > 0 && dep.Spec.Template.Annotations == nil {
+			return StrategyError{Reason: StrategyErrReasonAnnotationsMissing, Message: fmt.Sprintf("no annotations found on deployment")}
+		}
+		for key, value := range i.templateAnnotations {
+			if dep.Spec.Template.Annotations[key] != value {
```

Comment:
At a glance this looks like it has the potential to panic if an annotation we expect to exist on the deployment template does not.

Reply:
if the key doesn't exist, you get the zero value for the value, not a panic: https://play.golang.org/p/Se7UYpYnleR

Reply:
cool - my bad.